src/univalue/univalue_read.cpp

```diff
@@ -188,7 +188,7 @@ enum jtokentype getJsonToken(string& tokenVal, unsigned int& consumed,
                 case 't':  valStr += "\t"; break;
 
                 case 'u': {
-                    char buf[4] = {0,0,0,0};
+                    char buf[3];
```

Comment:
Exactly that was my rationale.  The buffer will only ever be accessed in the iterator range `[buf, last)`, and `last` itself is incremented at most three times.

However, I like your suggestion of using `push_back` directly!  I think that makes sense and will rewrite the code accordingly.